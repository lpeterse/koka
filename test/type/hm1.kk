fun id(x) {
  return x;
}

fun apply(f,x) {
  f(x)
}

fun test() {
  fun i(x) { x }
  i(i)
}

fun test2() {
  single(hm1/id)
}

fun test3() = choose(fun(x,y){x},fun(x,y){y})
fun test4(f) = choose(core/id,f)
