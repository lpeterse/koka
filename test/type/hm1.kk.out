apply     : forall<a,b,e> (f : (a) -> e b, x : a) -> e b
id        : forall<a> (x : a) -> a
test      : forall<a> () -> ((x : a) -> a)
test2     : forall<a,e> () -> list<(x : a) -> e a>
test3     : forall<a> () -> ndet ((x : a, y : a) -> a)
test4     : forall<a,e> (f : (x : a) -> e a) -> ndet ((x : a) -> e a)
